### PR TITLE
Add webserver WSGI path configuration

### PR DESCRIPTION
**Pull request by weiiwang01**
And a charm configuration called `webserver_wsgi_path` to allow the charm run with different Flask project structures.

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -58,3 +58,7 @@ options:
   webserver_workers:
     type: int
     description: The number of webserver worker processes for handling requests.
+  webserver_wsgi_path:
+    type: string
+    default: "app:app"
+    description: The WSGI application path. By default, it's set to "app:app".
```
Comment (arturo-seijas) [inline config.yaml]: Missing line at EOF

Comment (weiiwang01) [reply config.yaml]: Added, thanks!

**Review by arturo-seijas — DISMISSED** [review]
LGTM

File: config.yaml [hunk 2]
```diff
@@ -58,3 +58,7 @@ options:
   webserver_workers:
     type: int
     description: The number of webserver worker processes for handling requests.
+  webserver_wsgi_path:
```
Comment (gregory-schiano) [inline config.yaml:61]: I'm wondering if we shouldn't name this `webserver_entrypoint` as it's more generic/reusable and path could look like FS path

Comment (weiiwang01) [reply config.yaml:61]: WSGI path is a common technical term in the Flask domain, so I think it might be more understandable for Flask users.

Comment (gregory-schiano) [reply config.yaml:61]: Yeap but we're in the webserver module, which is meant to become generic

Comment (weiiwang01) [reply config.yaml:61]: I think the WSGI path is applicable for all Flask web servers, such as Gunicorn, uWSGI, etc.

Comment (gregory-schiano) [reply config.yaml:61]: The webserver doesn't aim to be Python specific, it aims to be generic to any 12-factor app charm as any web application requires a webserver
Using generic config allows to have a better user experience. We could keep this name for now, but we'll very likely rename it later, so it doesn't cost much to name is appropriately now

Comment (weiiwang01) [reply config.yaml:61]: The problem is that wsgi path is specify to the WSGI server, for other type of web servers such as fastcgi server, there's a correlating concept to the wsgi path in WSGI server. Maybe we can rename the `webserver` module to `wsgi_webserver` module?

Comment (weiiwang01) [reply config.yaml:61]: For other configurations, there are similar problems, for example, a cgi webserver doesn't have the concept of workers or threads.

File: src/charm_state.py [hunk 3]
```diff
@@ -115,11 +117,15 @@ def __init__(
                 or None if not specified.
             webserver_timeout: The request silence timeout for the web server,
                 or None if not specified.
+            webserver_wsgi_path: The WSGI application path, or None if not specified.
```
Comment (amandahla) [inline src/charm_state.py:121]: None or "app:app"?

Comment (weiiwang01) [reply src/charm_state.py:121]: The default value for the `webserver_wsgi_path` function argument is `None`. This is due to the typing of the ops framework. The `None` input will be interpreted as `"app:app"`.

File: config.yaml (hunk 2 above)
Comment (jdkandersson) [reply config.yaml:61]: The renaming later sounds like a good option when we have another example, e.g., the fastAPI or spring boot case

**Review by jdkandersson — DISMISSED** [review]

**Review by merkata — DISMISSED** [review]

**Review by amandahla — APPROVED** [review]
LGTM

**Review by jdkandersson — APPROVED** [review]

**Pull request merged**